climada/engine/test/test_supplychain.py

```diff
@@ -0,0 +1,180 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Test Supplychain class.
+"""
+
+import unittest
+import numpy as np
+
+from climada import CONFIG
+from climada.entity.exposures.base import Exposures
+from climada.entity import ImpactFuncSet, IFTropCyclone
+from climada.hazard.base import Hazard
+from climada.engine.supplychain import SupplyChain
+from climada.util.constants import EXP_DEMO_H5
+
+HAZ_TEST_MAT = CONFIG.hazard.test_data.dir().joinpath('atl_prob_no_name.mat')
+DIR_TEST_DATA = CONFIG.engine.test_data.dir()
+
+class TestSupplyChain(unittest.TestCase):
+    """Testing the SupplyChain class."""
+    def test_read_wiot(self):
+        """Test reading of wiod table."""
+        sup = SupplyChain()
+        sup.read_wiod(year='test', file_path=DIR_TEST_DATA)
+
+        self.assertAlmostEqual(sup.mriot_data[0, 0], 12924.1797, places=3)
+        self.assertAlmostEqual(sup.mriot_data[0, -1], 0, places=3)
+        self.assertAlmostEqual(sup.mriot_data[-1, 0], 0, places=3)
+        self.assertAlmostEqual(sup.mriot_data[-1, -1], 22.222, places=3)
+        
+        self.assertAlmostEqual(sup.mriot_data[0, 0], 
+                               sup.mriot_data[sup.cntry_pos[list(sup.cntry_pos)[0]][0], 
+                                              sup.cntry_pos[list(sup.cntry_pos)[0]][0]], 
+                               places=3)
+        self.assertAlmostEqual(sup.mriot_data[-1, -1], 
+                               sup.mriot_data[sup.cntry_pos[list(sup.cntry_pos)[-1]][-1], 
+                                              sup.cntry_pos[list(sup.cntry_pos)[-1]][-1]], 
+                               places=3)        
+        self.assertEqual(np.shape(sup.mriot_data), (112, 112))
+        self.assertAlmostEqual(sup.total_prod.sum(), 3533367.89439, places=3)
+        self.assertEqual(len(sup.countries_iso3), len(sup.countries))
+
+    def test_calc_impact(self):
```

Comment:
I am not sure about this, but I would think it is easier to debug if this function is separated into several functions which each test one single function of the code. @emanuel-schmid , could you please comment?

Reply:
Do you mean to separate `sup.calc_sector_direct_impact` from `sup.calc_indirect_impact`? Then I agree, it's best to write unit tests around exactly one method/function to be scrutinized.

Reply:
Yes but in oder to run  `sup.calc_indirect_impact` you need to have previosuly run `sup.calc_sector_direct_impact` that is why I put them in the same test

Reply:
I still would do separate test functions, otherwise, if it fails you do not know whether it is `sup.calc_indirect_impact` or `sup.calc_sector_direct_impact` that has a problem.